### DISCUSSION TITLE
Using host IP instead of dns name

### DISCUSSION THREAD

**Discussion by giuliohome** [kedacore/http-add-on#425] [Q&A] [answered]
Hi all and thank you for sharing.
So my question is if it is possible to use the ip of the service to scale the deployment, hence based on the http request to that ip instead of filtering with the dns name.
Any feedback in terms of tips regarding either the usage of the plugin or the development changes to do would be much appreciated.

**Comment by giuliohome** [comment, marked as answer]
At first sight, [this](https://github.com/kedacore/http-add-on/blob/c211da9bd1ec400d4d2316f20176655eddeea59e/interceptor/middleware.go#L12-L23) seems to be the point to be considered

```go
func getHost(r *nethttp.Request) (string, error) {
	// check the host header first, then the request host
	// field (which may contain the actual URL if there is no
	// host header)
	if r.Header.Get("Host") != "" {
		return r.Header.Get("Host"), nil
	}
	if r.Host != "" {
		return r.Host, nil
	}
	return "", fmt.Errorf("host not found")
}
```

I have to verify if it already contains the ip in case it was sent from the client (and if the add-on is already usable to filter by ip as it is now, in such a situation).

I'm already looking at `this` answers saying

> Per RFC 2616, this is either the value of the Host: header or the host name given in the URL itself.

and [this](https://stackoverflow.com/a/40583889) suggesting instead `req.URL.RequestURI()` (maybe in a fork if this is the way to go)

Well, I see in the interceptor's code comment itself that it reads:

> which may contain the actual URL if there is no host header

so I would guess there is no code change needed, though this fact was not apparent from the README.

Will [test it](https://github.com/giuliohome/kedacore-http-add-on/issues/1) soon...
... Done, OK (just notice that the ingress has been deleted if one has to go only with the IP address instead of a dns name)

**Reply by arschles** [reply]
@giuliohome I'm not sure if you have a question or you've figured it out on your own. Can you let me know?

**Reply by giuliohome** [reply]
Hi! Yes, my question is an actual question. In the meantime I tried to answer it on my own, but I would be happy to receive other answers if mine is wrong. Basically I'm saying that the constraint in using a dns name instead of an ip address is only given by the presence of an ingress in the helm chart.
Well, since I've actually tested 127.0.0.1:8080 as a working host, maybe an alternative (now [tested](https://github.com/giuliohome/kedacore-http-add-on/issues/2) and confirmed ok) would have been writing localhost as host... Aside from it, any feedback about my question and/or my own solution would be appreciated. Thanks
